### PR TITLE
2640 export initial conditions for expressions as part of the expression profile

### PR DESCRIPTION
**Pull request by rwmcintosh**

### CODE REVIEW

File: src/PKSim.Core/Services/SimulationConfigurationTask.cs [hunk 1]
```diff
@@ -107,15 +108,20 @@ public SimulationConfiguration CreateFor(Simulation simulation, bool shouldValid
          //STEP8: Add Observers
          module.Add(_modelObserverQuery.AllObserversFor(module.Molecules, simulation));
 
-         //STEP9 once all building blocks have been created, we need to create the default parameter and molecule values values 
-         var initialConditions = _initialConditionsCreator.CreateFor(module, simulation);
-         module.Add(initialConditions);
+         //STEP9 once all building blocks have been created, we need to create the default parameter and compound molecule values
+         var allDefinedProteins = module.Molecules.Where(x => simulation.Individual.AllDefinedMolecules().AllNames().Contains(x.Name)).ToList();
```
Comment (msevestre) [inline src/PKSim.Core/Services/SimulationConfigurationTask.cs:112]: Feels like simulation.Individual.AllDefinedMolecules().AllNames() is going to be retrieved alk the time

File: src/PKSim.Core/Services/SimulationConfigurationTask.cs [hunk 2]
```diff
@@ -107,15 +108,20 @@ public SimulationConfiguration CreateFor(Simulation simulation, bool shouldValid
...
+         var allOtherMolecules = module.Molecules.Except(allDefinedProteins).ToList();
```
Comment (msevestre) [inline src/PKSim.Core/Services/SimulationConfigurationTask.cs:113]: What about 2 PORES Molecules here .
@rwmcintosh Can you create a model using TWO PORES and make sure it behaves the same as in v11?

File: src/PKSim.Core/Services/SimulationConfigurationTask.cs [hunk 3]
```diff
@@ -125,5 +131,21 @@ public SimulationConfiguration CreateFor(Simulation simulation, bool shouldValid
...
+      }
+
+      private InitialConditionsBuildingBlock createInitialConditionsForCompounds(Module module, Simulation simulation, IReadOnlyList<MoleculeBuilder> compoundMolecules)
+      {
+         return _initialConditionsCreator.CreateFor(module.SpatialStructure, compoundMolecules, simulation);
+      }
+
+      private void addInitialConditions(ExpressionProfileBuildingBlock expressionProfile, SpatialStructure spatialStructure, IReadOnlyList<MoleculeBuilder> molecules, Simulation simulation)
+      {
+         var initialConditions = _initialConditionsCreator.CreateFor(spatialStructure, molecules.Where(molecule => Equals(molecule.Name, expressionProfile.MoleculeName)).ToList(), simulation);
```
Comment (msevestre) [inline src/PKSim.Core/Services/SimulationConfigurationTask.cs:147]: FindByName? 
Feels like we need an overload with only one molecule in _initialConditionsCreator .CreateFor

File: src/PKSim.Core/Snapshots/ExpressionProfile.cs [hunk 4]
```diff
@@ -7,17 +7,13 @@ namespace PKSim.Core.Snapshots
 {
    public class ExpressionProfile : SnapshotBase, IBuildingBlockSnapshot
    {
-      [Required]
-      public QuantityType Type { get; set; }
+      [Required] public QuantityType Type { get; set; }
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/ExpressionProfile.cs:10]: not sure about this change here. Can you revert?

File: tests/PKSim.UI.Starter/PKSim.UI.Starter.csproj [hunk 5]
```diff
@@ -56,14 +56,14 @@
...
-    <PackageReference Include="OSPSuite.Core" Version="12.0.0-zooXO" />
+  <PackageReference Include="OSPSuite.Assets" Version="12.0.0-ojEDY" />
+    <PackageReference Include="OSPSuite.Core" Version="12.0.0-ojEDY" />
     <PackageReference Include="OSPSuite.DevExpress" Version="21.2.3" />
     <PackageReference Include="OSPSuite.FuncParser" Version="4.0.0.54" GeneratePathProperty="true" />
     <PackageReference Include="OSPSuite.SimModel" Version="4.0.0.53" GeneratePathProperty="true" />
     <PackageReference Include="OSPSuite.SimModelSolver_CVODES" Version="4.1.0.8" GeneratePathProperty="true" />
     <PackageReference Include="System.Data.SQLite.Core" Version="1.0.112" GeneratePathProperty="true" />
-    <PackageReference Include="OSPSuite.Presentation" Version="12.0.0-zooXO" GeneratePathProperty="true" />
+    <PackageReference Include="OSPSuite.Presentation" Version="12.0.0-ojEDY" GeneratePathProperty="true" />
```
Comment (msevestre) [inline tests/PKSim.UI.Starter/PKSim.UI.Starter.csproj:66]: still some old version referenced here

**Comment by rwmcintosh** [comment]
> Looks good. Some small changes
> 
> Hard for me to tell what' snot working based on our convo

**Pull request merged**